Documentation/ApiOverview/FlexForms/Index.rst

```diff
@@ -0,0 +1,380 @@
+.. include:: ../../Includes.txt
+
+.. _flexforms:
+
+==============
+FlexForms
+==============
+
+FlexForms can be used to configure extension plugins in the backend.
+
+Credits
+=======
+
+Some of the examples were taken from the extensions
+`news <https://extensions.typo3.org/extension/example/>`__ (by Georg Ringer)
+and `bootstrap_package <https://extensions.typo3.org/extension/example/>`__
+(by Benjamin Kott). This chapter was originally written by Sybille Peters.
+
+Further enhancements by the TYPO3 community are welcome!
+
+When to Use FlexForms
+=====================
+
+Extensions can be configured using :ref:`Extension Configuration <extension-options>`.
+This configuration applies to the extension in general. You may want to configure
+individual plugins differently, depending on where they are added. The
+configuration set via the FlexForm mechanism applies to only the content
+record it has been configured for. The FlexForms configuration
+can be changed by editors. This gives editors more control over plugin features
+and what is to be rendered.
+
+FlexForms configuration has more possibilities than Extension Configuration.
+Extension Configuration uses TypoScript constant syntax. This limits you to one
+single line per configuration option with very little flexibility.
+
+With FlexForm, it is possible to populate select lists via functions
+(:ref:`itemsProcFunc <flexforms-itemsProcFunc>`),
+show options conditionally and more.
+
+Example Use Cases
+=================
+
+* Use the plugin to configure a different view, e.g. list, single
+* The `bootstrap_package <https://extensions.typo3.org/extension/example/>`__
+  uses FlexForms to configure rendering options,
+  e.g. a transition interval and transition type (slide, fade)
+  for the carousel.
+
+.. image:: Images/FlexFormCarousel.png
+   :class: with-shadow
+
+
+How it Works
+============
+
+#. In the extension, a configuration schema is defined and attached to one or more plugins.
+#. When the plugin is added to a page, it can be configured as defined by the configuration
+   schema.
+#. The configuration for this content element is automatically saved to `tt_content.pi_flexform`.
+#. The extension can read current configuration in the controller and act according to
+   the configuration for the current plugin.
+
+Steps to Perform (Extension Developer)
+======================================
+
+.. rst-class:: bignums-xxl
+
+#. Create configuration schema
+
+   A configuration schema in XML is added to the extension.
```

Comment:
I would point to TCA Reference and https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/DataFormats/T3datastructure/Index.html